### PR TITLE
fix(lib/grandpa): Duplicate votes is GRANDPA are counted as equivocatory votes (GSR-11)

### PR DESCRIPTION
**Pull request by edwardmack**
## Changes
- Implement fix to `getEquivocatoryVoters` that does not include duplicate votes as equivocatory votes, addressing Sigma Prime - Gossamer Audit issue 11 (GSR-11)
- Add unit tests for testing duplicate votes condition.

## Tests

```sh
go test -run ^Test_getEquivocatoryVoters$ github.com/ChainSafe/gossamer/lib/grandpa -v
```

## Issues
closes: #2410 

## Primary Reviewer

@kishansagathiya

### CODE REVIEW

File: lib/grandpa/message_handler.go [hunk 1]
```diff
@@ -284,22 +284,21 @@ func (h *MessageHandler) verifyCatchUpResponseCompletability(prevote, precommit
...
+func getEquivocatoryVoters(votes []AuthData) map[ed25519.PublicKeyBytes]struct{} {
 	eqvVoters := make(map[ed25519.PublicKeyBytes]struct{})
-	voters := make(map[ed25519.PublicKeyBytes]int, len(votes))
+	voters := make(map[ed25519.PublicKeyBytes][64]byte, len(votes))
 
 	for _, v := range votes {
-		voters[v.AuthorityID]++
-		switch voters[v.AuthorityID] {
-		case 1:
-		case 2:
```
Comment (kishansagathiya) [inline lib/grandpa/message_handler.go:295]: Does not this already take cared of the case when a vote appears twice in the AuthData list?

File: lib/grandpa/message_handler.go [hunk 2]
```diff
@@ -284,22 +284,21 @@ func (h *MessageHandler) verifyCatchUpResponseCompletability(prevote, precommit
...
-			eqvVoters[v.AuthorityID] = struct{}{}
-		default:
-			return nil, fmt.Errorf("%w: authority id %x has %d votes",
-				errInvalidMultiplicity, v.AuthorityID, voters[v.AuthorityID])
+		signature, present := voters[v.AuthorityID]
+		if present {
+			if !bytes.Equal(signature[:], v.Signature[:]) {
```
[the remark this answers is not in the document]
Comment (edwardmack) [reply lib/grandpa/message_handler.go:294]: If there is a vote present for the authority, and the signature of that vote doesn't equal the votes signature then we add that voter to the equivocatory voters map.  The voter is equivocatory since there are votes for that voter that do not match.

File: lib/grandpa/message_handler.go (hunk 1 above)
Comment (edwardmack) [reply lib/grandpa/message_handler.go:295]: The audit issue and spec say that duplicate votes are not equivocatory, just when votes don't match they are then equivocatory.

**Review by EclesioMeloJunior — CHANGES_REQUESTED** [review]
